numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
Ah, good point. It's still not entirely clear to me why the current duck typing solution dies not suffice, though.